app/src/main/java/com/douglasbruce/tasky/features/agenda/AgendaViewModel.kt

```diff
@@ -64,14 +72,53 @@ class AgendaViewModel @Inject constructor(
                     selectedDay = 0,
                     showDatePicker = false,
                 )
+                getAgendaForSelectedDate()
             }
 
             is AgendaEvent.OnDayClick -> {
                 state = state.copy(
                     selectedDay = event.day,
                     displayDate = state.selectedDate.plusDays(event.day.toLong())
                 )
+                getAgendaForSelectedDate()
+            }
+        }
+    }
+
+    private fun getAgendaForSelectedDate() {
+        viewModelScope.launch {
+            agendaRepository.getAgendaForDate(
+                state.selectedDate.plusDays(state.selectedDay.toLong())
+            ).collect { agendaItems ->
+                val item = getItemBeforeTimeNeedle(agendaItems)
+                state = state.copy(
+                    items = agendaItems.sortedBy { it.sortDate },
+                    itemBeforeTimeNeedle = item
+                )
+            }
+        }
```

Comment:
You can also collect flows without this extra indentation from `viewModelScope`:
```
flow.onEach {
    // Just like collect
}.launchIn(viewModelScope)
```